### DISCUSSION TITLE
Model-view does not work on Xiaomi and some samsung

### DISCUSSION THREAD

**Discussion by AlirezaRastgoo** [google/model-viewer#4446] [Q&A]
Sorry, my English is not very good
Since the update of Google Play Services for AR on September 4, 2023, Model-Viewer no longer works on Xiaomi devices and some Samsung devices. 
However, until September 3, 2023, Model-Viewer was working in WebXR mode on Xiaomi as well as Samsung devices. When I connected the device to my computer's Chrome, I got the following errors:

prismatic.min.js:75 Uncaught TypeError: document.registerElement is not a function
    at prismatic.min.js:75:96
    at prismatic.min.js:21:156
    at prismatic.min.js:21:175

Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "model-viewer" has already been used with this registry
    at https://ajax.googleapis.com/ajax/libs/model-viewer/3.1.1/model-viewer.min.js:1094:30002
    
    
And after click on Show AR , I got the following errors:
    
Attempting to present in AR with WebXR...

Could not create a session because: Something went wrong initializing the session in the runtime

ar.js:208 Error while trying to present in AR with WebXR
[enterARWithWebXR] @ ar.js:208
await in [enterARWithWebXR] (async)
activateAR @ ar.js:145
ARModelViewerElement.<computed> @ ar.js:75

ar.js:209 DOMException: The specified session configuration is not supported.
[enterARWithWebXR] @ ar.js:209
await in [enterARWithWebXR] (async)
activateAR @ ar.js:145
ARModelViewerElement.<computed> @ ar.js:75

ARRenderer.js:254 Error while trying to end WebXR AR session

**Comment by elalish** [comment]
> Failed to execute 'define' on 'CustomElementRegistry': the name "model-viewer" has already been used with this registry

This looks like the primary problem. It appears you're loading our library twice (or maybe two versions of it?). I'm guessing if you track that down, the rest may resolve.

**Reply by kkorniszuk** [reply]
The issue seems to still occur when `model-viewer` is loaded twice in the application's source code — for example, once directly and once from a dependency that includes its own version of `model-viewer`. I can create a pull request to address this.
